nautobot_design_builder/tests/test_changes.py

```diff
@@ -0,0 +1,37 @@
+"""Tests for change utilities."""
+
+from unittest import TestCase
+
+from nautobot_design_builder.changes import revert_changed_dict
+
+
+class TestRevertChangedDict(TestCase):
+    """TestCase for the revert_changed_dict utility."""
+
+    def test_added_key(self):
+        original = {"key1": "initial-value"}
+        changed = {"key1": "initial-value", "key2": "new-value"}
+        current = {**changed}
+        got = revert_changed_dict(current, original, changed)
+        self.assertDictEqual(original, got)
+
+    def test_extra_key(self):
+        original = {"key1": "initial-value"}
+        changed = {"key1": "initial-value", "key2": "new-value"}
+        current = {"key1": "initial-value", "key2": "new-value", "key3": "key3-value"}
+        got = revert_changed_dict(current, original, changed)
+        self.assertDictEqual({"key1": "initial-value", "key3": "key3-value"}, got)
+
+    def test_missing_changed_value_without_current(self):
+        original = {"key1": "initial-value"}
+        changed = {"key2": "new-value"}
+        current = {"key2": "new-value", "key3": "key3-value"}
+        got = revert_changed_dict(current, original, changed)
+        self.assertDictEqual({"key1": "initial-value", "key3": "key3-value"}, got)
```

Comment:
this raises an interesting point. 
I would like to have the option to only define a few keys of the dict, so the keys not impacted by the design (e.g. key1),I don't care about their value. Only the keys added by the design (e.g. key2) have to be moved back to previous state, either removed or changed.
But, on the other side, I would like to have a way to define the whole dict as a change, not only targeting a few of the keys.
Both options look like two different operations... how do you see this?

Reply:
@abates , I saw you merged it without resolving this comment. does it mean I shall raise it as an issue?

Reply:
You are right, I got confused with the behavior